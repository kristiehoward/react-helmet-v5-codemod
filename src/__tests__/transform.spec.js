const defineTest = require('../../test/testUtils').defineTest;

defineTest(__dirname, 'transform', null, 'HelmetReadme', null);
defineTest(__dirname, 'transform', null, 'SelfClosing', null);
defineTest(__dirname, 'transform', null, 'VariableAndFunction', null);
defineTest(__dirname, 'transform', null, 'SpreadProps', 'Cannot process JSX spread attributes');
defineTest(__dirname, 'transform', null, 'InvalidProps', null);
