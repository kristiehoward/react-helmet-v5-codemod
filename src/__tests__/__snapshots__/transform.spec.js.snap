// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform transforms correctly using "HelmetReadme" data 1`] = `
"import React from \\"react\\";
import Helmet from \\"react-helmet\\";

// Previous react-helmet readme before v5.0.0
// https://github.com/nfl/react-helmet/blob/f392aaef69cce3cd1951dd96e4f860bf8104843a/README.md

export default function Application () {
    return (
        <div className=\\"application\\">
            <Helmet
                titleTemplate=\\"MySite.com - %s\\"
                defaultTitle=\\"My Default Title\\"
                onChangeClientState={(newState) => console.log(newState)}>
                <html lang=\\"en\\" amp />
                <title itemprop=\\"name\\" lang=\\"en\\">My Title</title>
                <base target=\\"_blank\\" href=\\"http://mysite.com/\\" />
                <meta name=\\"description\\" content=\\"Helmet application\\" />
                <meta property=\\"og:type\\" content=\\"article\\" />
                <link rel=\\"canonical\\" href=\\"http://mysite.com/example\\" />
                <link
                    rel=\\"apple-touch-icon\\"
                    href=\\"http://mysite.com/img/apple-touch-icon-57x57.png\\" />
                <link
                    rel=\\"apple-touch-icon\\"
                    sizes=\\"72x72\\"
                    href=\\"http://mysite.com/img/apple-touch-icon-72x72.png\\" />
                <script src=\\"http://include.com/pathtojs.js\\" type=\\"text/javascript\\" />
                <script type=\\"application/ld+json\\">{\`{ \\"@context\\": \\"http://schema.org\\" }\`}</script>
                <noscript>{\`<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"foo.css\\" />\`}</noscript>
                <style type=\\"text/css\\">{\\"body {background-color: blue;} p {font-size: 12px;}\\"}</style>
            </Helmet>
        </div>
    );
};"
`;

exports[`transform transforms correctly using "InvalidProps" data 1`] = `
"import React from \\"react\\";
import Helmet from \\"react-helmet\\";

// Put some invalid (unrecognized) props on Helmet that should get stripped out
export default function Application () {
    return (
        <div className=\\"application\\">
            <Helmet titleTemplate=\\"MySite.com - %s\\" defaultTitle=\\"My Default Title\\" />
        </div>
    );
};"
`;

exports[`transform transforms correctly using "SelfClosing" data 1`] = `
"import React from \\"react\\";
import Helmet from \\"react-helmet\\";

// The resulting <Helmet> tag should be self closing
export default function Application () {
    return (
        <div className=\\"application\\">
            <Helmet titleTemplate=\\"MySite.com - %s\\" defaultTitle=\\"My Default Title\\" />
        </div>
    );
};"
`;

exports[`transform transforms correctly using "VariableAndFunction" data 1`] = `
"import React, { Component } from 'react';
import { connect } from 'react-redux';

export default class Root extends Component {
  static propTypes = {
    children: node.isRequired,
    location: object.isRequired,
    params: object.isRequired,
  }

  render() {
    // Pick meta tags based on environment
    const meta = !isProd() ?
      [{ name: 'robots', content: 'noindex' }] : [];
    // Some function that could be running
    const getScripts = () => { return [{}] };
    return (
      <div>
        <Helmet titleTemplate={'%s - Docker Store'}>
          {/*WARNING: Unable to translate the following tag*/
          }<meta>{meta}</meta>
          {/*WARNING: Unable to translate the following tag*/
          }<script>{getScripts()}</script>
          <title>{\\"123\\" === \\"123\\" ? \\"Something\\" : \\"Another Thing\\"}</title>
        </Helmet>
        <div id=\\"mainContainer\\" className={mainClasses}>
          {this.props.children}
        </div>
      </div>
    );
  }
}"
`;
